v2/devices/FP4.yml

```diff
@@ -0,0 +1,190 @@
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection.'
+    image: "phone_power_up"
+    button: true
+  fastbootd:
+    title: "Reboot to fastbootd mode"
+    description: 'With the device still at the "FastBoot Mode", use the Volume buttons to switch to "Recovery Mode" and push the power button to confirm your selection. Then select "Fastboot mode" from the list.'
+    image: "phone_power_up"
+    button: true
+unlock:
+  - "confirm_model"
+  - "confirm_os"
+  - "unlock_phone"
+# FIXME bootloader_locked? https://www.fairphone.com/en/bootloader-unlocking-code-for-fairphone-3
+operating_systems:
+  - name: "Ubuntu Touch"
+    compatible_installer: ">=0.9.6-beta"
+    options:
+      - var: "channel"
+        name: "Channel"
+        tooltip: "The release channel"
+        link: "https://docs.ubports.com/en/latest/about/process/release-schedule.html"
+        type: "select"
+        remote_values:
+          systemimage:channels:
+      - var: "wipe"
+        name: "Wipe Userdata"
+        tooltip: "Wipe personal data. This is necessary for first time installations."
+        type: "checkbox"
+      - var: "partition"
+        name: "Partitioning"
+        tooltip: "Changes partition sizes as required. Only necessary for first time installations."
+        type: "checkbox"
+        value: true
+      - var: "bootstrap"
+        name: "Bootstrap"
+        tooltip: "Flash system partitions using fastboot"
+        type: "checkbox"
+        value: true
+    prerequisites: []
+    steps:
+      ### Ensure we always start in bootloader mode
+      - actions:
+          - adb:reboot:
+              to_state: "bootloader"
+        fallback:
+          - core:user_action:
+              action: "bootloader"
+      ### As this is an A/B device, force all future operations in "a" slot.
+      - actions:
+          - fastboot:set_active:
+              slot: "a"
+      ### Flash important partitions (if requested)
+      - actions:
+          - core:download:
+              group: "firmware"
+              files:
+                # FIXME cdimage - also maybe compress if possible? Should decrease boot/recovery.img from 96MB to ~24MB
+                - url: "https://files.beidl.guru/fp4/boot.img"
+                  checksum:
+                    sum: "c1ab7a12e48edddc089cd93265e475ff58242919427520a589fc19203c8fb628"
+                    algorithm: "sha256"
+                - url: "https://files.beidl.guru/fp4/dtbo.img"
+                  checksum:
+                    sum: "68ad83e20eb2b47592c2b7fe48161893770b1727c7b7ce7240504eaf8612d733"
+                    algorithm: "sha256"
+                - url: "https://public.z3ntu.xyz/tmp/recovery-fastbootd.img"
+                  checksum:
+                    sum: "ae7c97ca17fa93cfefd20c25e09d80f03b4921fe30f62f157ef85702a892ed53"
+                    algorithm: "sha256"
```

Comment:
These images should be hosted on cdimage, and maybe built on a CI system if possible?

Anyone know if they can be compressed and unpacked by the installer? Both boot and recovery partitions are 96MB on this device and therefore contain a lot of free space